synapse/server_notices/server_notices_manager.py

```diff
@@ -288,3 +325,51 @@ async def _update_notice_user_profile_if_changed(
                 ratelimit=False,
                 content={"displayname": display_name, "avatar_url": avatar_url},
             )
+
+    async def _update_room_info(
+        self,
+        requester: Requester,
+        room_id: str,
+        info_event_type: str,
+        info_content_key: str,
+        info_value: Optional[str],
+        if_changed: Optional[bool] = True,
+    ) -> None:
+        """
+        Updates a specific notice room's info if it's different from what is set.
+
+        Args:
+            requester: The user who is performing the update.
+            room_id: The ID of the server notice room
+            info_event_type: The event type holding the specific info
+            info_content_key: The key containing the specific info in the event's content
+            info_value: The expected value for the specific info
+            if_changed: Try to read the info first and does not send the event if it's already the same
+        """
+        if if_changed:
+            room_info_event = await self._message_handler.get_room_data(
+                requester,
+                room_id,
+                info_event_type,
+                "",
+            )
+
+            if room_info_event and room_info_event.get(info_content_key) == info_value:
```

Comment:
This won't handle the case where we've set it to the empty string? Something like:

```suggestion
            existing_info_value = None
            if room_info_event:
                existing_info_value = room_info_event.get(info_content_key)
            if existing_info_value == info_value:
                return
            if not existing_info_value and not info_value:
                # A missing `info_value` can either be represented by a None
                # or an empty string, so we assume that if they're both falsey
                # they're equivalent.
                return
```